### DISCUSSION TITLE
Cant run a newly created project

### DISCUSSION THREAD

**Discussion by rotleaf** [fzyzcjy/flutter_rust_bridge#2426] [Q&A] [answered]
after creating a new project using `flutter_rust_bridge_codegen create fz` , and using flutter run, the compilation process errors with the following message

```
SEVERE: error: linking with `cc` failed: exit status: 1
SEVERE:   |
SEVERE:   = note: LC_ALL="C" PATH="/home/mbithi/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin:
```

followed all quickstart guides but it seems to never compile past this, I am trying to compile for an android emulator 'x86_64-linux-android', any ideas? has anyone else faced the same? I am on fedora 41 workstation linux

**Comment by fzyzcjy** [comment, marked as answer]
That looks like problem related to Rust compilation, i.e. Cargokit or Rust, instead of flutter_rust_bridge. But you can search through frb's issue section, e.g. https://github.com/search?q=repo%3Afzyzcjy%2Fflutter_rust_bridge+linking+failed&type=issues, and see whether something solves the problem. For example, a common one: https://github.com/fzyzcjy/flutter_rust_bridge/issues/1753

**Reply by rotleaf** [reply]
it's mainly because I was using flutter from snap, installing the official flutter fixed it, maybe it's worth mentioning this to linux users, another problem hits me. with external crates, Reqwest, openssl and hyper don't seem to compile for android devices, and also, which is the preferred ndk version to compile the rust binary?

**Reply by fzyzcjy** [reply]
Looks reasonable! I will update the doc in the next batch.

> with external crates, Reqwest, openssl and hyper don't seem to compile for android devices, and also, which is the preferred ndk version to compile the rust binary?

I have seen some questions about those in the issue section, so maybe also search there. Indeed it is not a bug of flutter_rust_bridge, but more like question "how to compile in android" for the specific Rust package.

**Reply by fzyzcjy** [reply]
Looks like it is already in doc: https://cjycode.com/flutter_rust_bridge/manual/troubleshooting#issues-with-glibc-eg-version-glibc_233-not-found

**Reply by rotleaf** [reply]
yep, its in docs, thanks

**Reply by rotleaf** [reply]
instead of reqwest, hyper or openssl, one could use raw TCP requests, those work fine

**Reply by fzyzcjy** [reply]
> yep, its in docs, thanks

You are welcome!

> instead of reqwest, hyper or openssl, one could use raw TCP requests, those work fine

Also looks like a alternative solution when one needs raw TCP
